core/src/main/java/org/elasticsearch/plugins/InstallPluginCommand.java

```diff
@@ -207,6 +211,24 @@ void execute(Terminal terminal, String pluginId, boolean isBatch, Map<String, St
             Files.createDirectory(env.pluginsFile());
         }
 
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
```

Comment:
I'm wondering if we should have the hook be added by the main method for Command, so that we can test. For example, have the Command be Closeable, and in close rm the files we have stashed?